cmake/DD4hepConfig.cmake.in

```diff
@@ -50,6 +52,10 @@ endif()
 FIND_DEPENDENCY(Boost REQUIRED)
 DD4HEP_SETUP_BOOST_TARGETS()
 
+if(DD4HEP_BUILD_DEBUG)
+  add_compile_definitions(DD4HEP_DEBUG=1)
```

Comment:
YES. Absolutely. 
Everything depending on DDCore (better including DD4hep/detail/ConditionsInterna.h) needs this flag.
Otherwise you have conflicting object layouts.